source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -1126,6 +1127,253 @@ namespace AdaptiveCards { namespace Rendering { namespace Uwp
         THROW_IF_FAILED(textConfig->get_MaxWidth(&maxWidth));
         StyleXamlTextBlock(textSize, textColor, containerStyle, Boolify(isSubtle), wrap, maxWidth, textWeight, xamlTextBlock, hostConfig);
     }
+    
+    HRESULT GetTextFromXmlNode(
+        ABI::Windows::Data::Xml::Dom::IXmlNode* node,
+        HSTRING* text)
+    {
+        ComPtr<ABI::Windows::Data::Xml::Dom::IXmlNode> localNode = node;
+        ComPtr<ABI::Windows::Data::Xml::Dom::IXmlNodeSerializer> textNodeSerializer;
+        RETURN_IF_FAILED(localNode.As(&textNodeSerializer));
+
+        RETURN_IF_FAILED(textNodeSerializer->get_InnerText(text));
+        return S_OK;
+    }
+
+    HRESULT XamlRunFromXmlNode(
+        ABI::Windows::Data::Xml::Dom::IXmlNode* node,
+        ABI::Windows::UI::Xaml::Documents::IRun** run)
+    {
+        HString text;
+        RETURN_IF_FAILED(GetTextFromXmlNode(node, text.GetAddressOf()));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IRun> localRun = XamlHelpers::CreateXamlClass<ABI::Windows::UI::Xaml::Documents::IRun>(HStringReference(RuntimeClass_Windows_UI_Xaml_Documents_Run));
+        RETURN_IF_FAILED(localRun->put_Text(text.Get()));
+
+        *run = localRun.Detach();
+        return S_OK;
+    }
+
+    HRESULT AddListInlines(
+        ABI::Windows::Data::Xml::Dom::IXmlNode* node,
+        bool isListOrdered,
+        IVector<ABI::Windows::UI::Xaml::Documents::Inline*>* inlines)
+    {
+        ComPtr<ABI::Windows::Data::Xml::Dom::IXmlNode> listChild;
+        RETURN_IF_FAILED(node->get_FirstChild(&listChild));
+
+        UINT iteration = 1;
+        while (listChild != nullptr)
+        {
+            HString text;
+            RETURN_IF_FAILED(GetTextFromXmlNode(listChild.Get(), text.GetAddressOf()));
+
+            std::wstring listElementString = L"\n";
+            if (!isListOrdered)
+            {
+                listElementString += L"● ";
+            }
+            else
+            {
+                wchar_t buffer[10];
+                swprintf_s(buffer, ARRAYSIZE(buffer), L"%d. ", iteration);
```

Comment:
iteration is user defined. ordered list can have user defined starting number

Reply:
Fixed this! Do we support the "type" attribute at all?

---
In reply to: [167377890](https://github.com/Microsoft/AdaptiveCards/pull/1113#discussion_r167377890) [](ancestors = 167377890)

Reply:
i'm not sure what you mean by "type"

---
In reply to: [167641053](https://github.com/Microsoft/AdaptiveCards/pull/1113#discussion_r167641053) [](ancestors = 167641053,167377890)

Reply:
It's an attribute on an ordered list that says whether the list should be, for example, 1,2,3,4 or a,b,c,d or I,II,III,IV. If you're not aware of it i'm guessing that means our markdown parser doesn't emit it :)

---
In reply to: [167732578](https://github.com/Microsoft/AdaptiveCards/pull/1113#discussion_r167732578) [](ancestors = 167732578,167641053,167377890)